compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -5487,7 +5488,7 @@ private BType getStructuredBindingPatternType(BLangVariable bindingPatternVariab
                         env.enclPkg.symbol.pkgID, fieldType, recordSymbol);
 
                 //TODO check below field position
-                fields.add(new BField(fieldName, bindingPatternVariable.pos, fieldSymbol));
+                fields.put(fieldName.value, new BField(fieldName, bindingPatternVariable.pos, fieldSymbol));
```

Comment:
Can't we use `fieldNameStr`?

```suggestion
                fields.put(fieldNameStr, new BField(fieldName, bindingPatternVariable.pos, fieldSymbol));
```

Reply:
Can, but doesn't make much of a difference right? Used `fieldName.value` since `fieldName` is what is used when creating the field.